plugins/huggingface/newhelm/suts/huggingface_client.py

```diff
@@ -250,18 +258,27 @@ def __init__(self, pretrained_model_name_or_path: str, **kwargs):
             self.device: str = "cuda:0"
         else:
             self.device = "cpu"
-        hg_kwargs = _process_huggingface_client_kwargs(kwargs)
+        self.hg_kwargs = _process_huggingface_client_kwargs(kwargs)
         self.model_path = pretrained_model_name_or_path
+        # Model and tokenizer are lazy loaded.
+        self.model: Optional[Any] = None
+        self.wrapped_tokenizer: Optional[WrappedPreTrainedTokenizer] = None
+
+    def _load_model(self) -> Tuple[Any, WrappedPreTrainedTokenizer]:
+        hugging_face_token = SECRETS.get_optional(scope="hugging_face", key="token")
+        if hugging_face_token is not None:
+            os.environ["HF_TOKEN"] = hugging_face_token
```

Comment:
If you wanted to avoid setting `HF_TOKEN` manually, I believe the `from_pretrained` methods for both the model and tokenizer take a `token` parameter. I don't see any big issue with setting `HF_TOKEN`, but wanted to mention it in case you were not aware.

Reply:
I'd waffled between the two, but I think I like `token` more than using `HF_TOKEN`. Updated.